snuba/datasets/plans/query_plan.py

```diff
@@ -67,14 +43,18 @@ def execute(
         raise NotImplementedError
 
 
-class ClickhouseQueryPlanBuilder(ABC):
+TQueryPlan = TypeVar("TQueryPlan")
```

Comment:
Wondering whether you thought about making the entire QueryPlan class generic instead? It does not appear to be highly ClickHouse specific as it is.